addons/hw_drivers/websocket_client.py

```diff
@@ -41,16 +41,30 @@ def send_to_controller(device_type, params):
 def on_message(ws, messages):
     """
         When a message is receive, this function is triggered
-        The message is load and if its type is 'iot_action', is sent to the device
+        The message is load and if its type is 'iot_device_action', is sent to the device
+        If the message type is 'iot_box_action', an action is executed on the box directly
     """
     messages = json.loads(messages)
     for message in messages:
-        if message['message']['type'] == 'iot_action':
-            payload = message['message']['payload']
-            if helpers.get_mac_address() in payload['iotDevice']['iotIdentifiers']:
-                for device in payload['iotDevice']['identifiers']:
-                    if device['identifier'] in main.iot_devices:
-                        main.iot_devices[device["identifier"]].action(payload)
+        message_type = message['message']['type']
+        if message_type == 'iot_device_action':
+            on_device_action_message(message)
+        elif message_type == 'iot_box_action':
+            on_box_action_message(message)
```

Comment:
assuming we will add more and more command in the future, maybe it would be smarter to refactor this system.

1. The simplest would be to use a dictionary that maps command string to a function.
2. But it might be smarter to do something more easy to extend like having an object with functions and use `getattr` and passing it the message. Like this, whatever function you add to the object, it will directly be callable if you send a message with the same string (a bit like route for http)